src/convenience/keyboard.ts

```diff
@@ -265,8 +497,38 @@ export class Keyboard {
     build() {
         return this.keyboard;
     }
+    /**
+     * Turns a two-dimensional keyboard button array into a keyboard instance.
+     * You can use the static button builder methods to create keyboard button
+     * objects.
+     *
+     * @param source A two-dimensional button array
+     * @param options Optional options for the custom keyboard
+     */
+    static from(
+        source: KeyboardSource,
+        options: Omit<ReplyKeyboardMarkup, "keyboard"> = {},
+    ): Keyboard {
+        function toButton(btn: KeyboardButtonSource) {
+            return typeof btn === "string" ? Keyboard.text(btn) : btn;
+        }
+        if (source instanceof Keyboard) {
+            source = source.clone();
+        } else {
+            source = new Keyboard(source.map((row) => row.map(toButton)));
+        }
+        return Object.assign(source, options);
+    }
 }
 
+type InlineKeyboardButtonSource =
+    | string
+    | [text: string, data?: string]
```

Comment:
i'd prefer we don't introduce another shortcut

Reply:
Drop the pairs? I'm fine with that, they're a bit of a hack anyway. So this?
- custom
  - string
  - button
- inline
  - string with data=label
  - button

Both as 1D/2D

Reply:
i can't see any use for string with data=label but it's fine, at least no one uses them

Reply:
I've been wanting to drop the default for data for 2.0. But it makes no sense to support it in one place and not support in in the other place. How about we only support buttons here and then indeed remove the default data for 2.0?

Reply:
having another array isn't that much better than `{text, data}` so… Having it on my own I wouldn't use the type so… guess it should be just removed.

Reply:
Addressed.